app/scripts/modules/kubernetes/src/v2/manifest/selector/labelEditor/LabelEditor.tsx

```diff
@@ -0,0 +1,126 @@
+
+export default class LabelEditor extends React.Component<ILabelEditorProps> {
+  public static defaultProps: Partial<ILabelEditorProps> = {
+    labelSelectors: [],
+    onLabelSelectorsChange: noop,
+  };
+
+  private static convertValueStringToArray = (value: string): string[] => {
+    return value.split(',').map((v: string) => v.trim());
+  };
+
+  private handleChange = (idx: number, property: keyof IManifestLabelSelector, value: string | string[]): void => {
+    this.props.onLabelSelectorsChange(
+      this.props.labelSelectors.map((ls, i) => {
+        if (idx !== i) {
+          return ls;
+        }
+        return {
+          ...ls,
+          [property]: value,
+        };
+      }),
+    );
+  };
+
+  private removeField = (idx: number): void => {
+    this.props.onLabelSelectorsChange(
+      this.props.labelSelectors.filter((_ls, i) => {
+        return idx !== i;
+      }),
+    );
+  };
+
+  private addField = (): void => {
+    return this.props.onLabelSelectorsChange(
+      this.props.labelSelectors.concat([
+        {
+          key: '',
+          kind: 'EQUALS',
+          values: [],
+        },
+      ]),
+    );
+  };
+
+  public render() {
+    return (
+      <table className="table table-condensed packed tags">
+        <thead>
+          <tr>
+            <th>Key</th>
+            <th>Kind</th>
+            <th>Value(s)</th>
+            <th />
+          </tr>
+        </thead>
+        <tbody>
+          {this.props.labelSelectors.map((selector, idx) => (
+            <tr className="label-editor-selector-row" key={idx}>
+              <td>
+                <input
+                  className="form-control input input-sm label-editor-key-input"
+                  type="text"
+                  value={selector.key}
+                  onChange={(e: any) => this.handleChange(idx, 'key', e.target.value)}
+                />
+              </td>
+              <td>
+                <Select
+                  className="label-editor-kind-select"
+                  clearable={false}
+                  onChange={(option: Option<string>) => this.handleChange(idx, 'kind', option.value)}
+                  options={LabelKeyOptions}
+                  value={selector.kind}
+                />
+              </td>
+              <td>
+                <input
+                  className="form-control input input-sm label-editor-values-input"
+                  onChange={(e: any) =>
+                    this.handleChange(idx, 'values', LabelEditor.convertValueStringToArray(e.target.value))
+                  }
+                  placeholder="Comma-separated values"
+                  value={selector.values.join(', ')}
+                  type="text"
+                />
+              </td>
+              <td>
+                <div className="form-control-static">
+                  <a className="label-editor-remove" onClick={() => this.removeField(idx)}>
```

Comment:
you can use <button className="link"> if you prefer. That way you don't have to use click event on <a>

Reply:
ah sorry I am not understanding, use what if I prefer?

Reply:
Forgot to quote it :D

Reply:
Oh good call, I kinda blindly copied the html from the angular template. Updated!